packages/client/src/components/Modals/SavedSearchPanel.vue

```diff
@@ -89,21 +91,34 @@ export default {
       deleteSavedSearchAPI: 'grants/deleteSavedSearch',
       fetchSavedSearches: 'grants/fetchSavedSearches',
       changeSelectedSearchId: 'grants/changeSelectedSearchId',
+      applyFilters: 'grants/applyFilters',
     }),
     setup() {
       this.fetchSavedSearches();
     },
     editSavedSearch(e) {
-      this.changeSelectedSearchId(e.target.getAttribute('searchid'));
+      const searchId = e.target.getAttribute('searchid');
+      // this.changeSelectedSearchId();
       this.$root.$emit('bv::toggle::collapse', 'saved-search-panel');
-      this.$root.$emit('bv::toggle::collapse', 'search-panel');
+      this.$emit('edit-filter', searchId);
     },
     deleteSavedSearch(e) {
       const searchId = `${e.target.getAttribute('searchid')}`;
       this.deleteSavedSearchAPI({ searchId });
       this.$root.$emit('bv::toggle::collapse', 'saved-search-panel');
       this.fetchSavedSearches();
     },
+    appylySavedSearch(searchId) {
+      const searchData = this.savedSearches.data.find((search) => search.id === searchId);
+      this.changeSelectedSearchId(searchId);
+      this.applyFilters(JSON.parse(searchData.criteria));
```

Comment:
Admittedly ugly; would be better to only have one parent action/mutation/event to trigger with the current search ID, and make it responsible for setting filters. Will look to refactor later on.